src/analyses/ai.h

```diff
@@ -140,51 +153,57 @@ class ai_baset
     output(ns, goto_function.body, "", out);
   }
 
-
+  /// Output the domains for the whole program as JSON
   virtual jsont output_json(
     const namespacet &ns,
     const goto_functionst &goto_functions) const;
 
+  /// Output the domains for a whole program as JSON
   jsont output_json(
     const goto_modelt &goto_model) const
   {
     const namespacet ns(goto_model.symbol_table);
     return output_json(ns, goto_model.goto_functions);
   }
 
+  /// Output the domains for a single function as JSON
   jsont output_json(
     const namespacet &ns,
     const goto_programt &goto_program) const
   {
     return output_json(ns, goto_program, "");
   }
 
+  /// Output the domains for a single function as JSON
   jsont output_json(
     const namespacet &ns,
     const goto_functionst::goto_functiont &goto_function) const
   {
     return output_json(ns, goto_function.body, "");
   }
 
-
+  /// Output the domains for the whole program as XML
   virtual xmlt output_xml(
     const namespacet &ns,
     const goto_functionst &goto_functions) const;
 
+  /// Output the domains for the whole program as XML
   xmlt output_xml(
     const goto_modelt &goto_model) const
   {
     const namespacet ns(goto_model.symbol_table);
     return output_xml(ns, goto_model.goto_functions);
   }
 
+  /// Output the domains for a single function as XML
   xmlt output_xml(
     const namespacet &ns,
     const goto_programt &goto_program) const
   {
     return output_xml(ns, goto_program, "");
   }
 
+  /// Output the domains for a single function as XML
```

Comment:
:pick: Might be better to just `\ref` to one so a little more detail can be provided about the format.

Reply:
If I had more to say I would consider using `\copydoc`, or manually doing that by saying "See \ref blah" (but one problem with that is that all these functions have the same name so I'd have to work out how to specify the one I want). But I don't have any more to say about this function at the moment. I'm mainly copying the comment for all of them so that it appears in the generated web pages.